samcli/commands/_utils/options.py

```diff
@@ -372,90 +372,17 @@ def _space_separated_list_func_type(value):
 _space_separated_list_func_type.__name__ = "LIST"
 
 
-def project_type_click_option(include_build=True):
-    return click.option(
-        "--project-type",
-        help="Project Type",
-        is_eager=True,
-        callback=partial(project_type_callback, include_build=include_build),
-        type=click.Choice(ProjectTypes.__members__, case_sensitive=False),
-    )
-
-
-def project_type_callback(ctx, param, provided_value, include_build):
+def plugin_common_options(f):
     """
-    Callback for `--project-type`
-    """
-    detected_project_type = determine_project_type(include_build)
-    if provided_value is not None:
-        if provided_value != detected_project_type:
-            raise click.BadOptionUsage(
-                option_name=param.name,
-                ctx=ctx,
-                message=f"It seems your project type is {detected_project_type}. "
-                f"However, you specified {provided_value} in --project-type",
-            )
-        LOG.debug("Using customized project type %s.", provided_value)
-        return provided_value
-
-    return detected_project_type
-
-
-def determine_project_type(include_build):
-    """
-    Determine the type of IaC Project to use.
-    If SAM template file exists, project_type will be “CFN”
-    Else if cdk.json exists in the root directory, project_type will be “CDK”
-    Else, project_type will be “CFN”
-
-    :param include_build: A boolean to set whether to search build template or not.
-    :return: Project type
-    """
-    LOG.debug("Determining project type...")
-
-    if find_cfn_template(include_build):
-        LOG.debug("The project is a CFN project.")
-        return ProjectTypes.CFN.value
-
-    if find_cdk_file():
-        LOG.debug("The project is a CDK project.")
-        return ProjectTypes.CDK.value
-
-    return ProjectTypes.CFN.value
-
-
-def find_in_paths(search_paths):
-    for path in search_paths:
-        if os.path.exists(path):
-            return True
-    return False
-
-
-def find_cfn_template(include_build):
-    """
-    Determine if template file exists
-    """
-    search_paths = ["template.yaml", "template.yml"]
-
-    if include_build:
-        search_paths.insert(0, os.path.join(".aws-sam", "build", "template.yaml"))
-
-    return find_in_paths(search_paths)
-
-
-def find_cdk_file():
-    """
-    Determine if cdk.json exists in the root directory
-    """
-    search_paths = ["cdk.json"]
-    return find_in_paths(search_paths)
-
-
-def cdk_click_options(f):
-    """
-    click options for specifying cdk related behavior
+    click options specific for cdk or cfn project type , and common options for both types of projects
```

Comment:
According to the discussion in our design, for now we want to statically put all plugin options (common ones, and cdk/cfn specific) into the option decorators together without splitting them in the interface .
However, I updated this part to make the CDK/CFN options more clear so that it can be divided and used better in the future.